packages/styles/scss/components/dialog/_dialog.scss

```
@@ -13,130 +13,171 @@
 @use '../../spacing' as *;
 @use '../../theme' as *;
 @use '../../type' as *;
-@use '../../utilities/ai-gradient' as *;
 @use '../../utilities/convert';
 @use '../../utilities/component-reset';
 @use '../../utilities/focus-outline' as *;
 @use '../../utilities/high-contrast-mode' as *;
+@use '../../utilities/update_fields_on_layer' as *;
+@use '../../utilities/z-index' as *;
 
 /// Dialog styles
 /// @access public
 /// @group dialog
 @mixin dialog {
   .#{$prefix}--dialog {
+    position: fixed;
+    z-index: z('modal');
+    display: flex;
+    align-items: center;
+    justify-content: center;
     padding: 0;
     border: none;
-    background-color: $layer;
-    block-size: fit-content;
+    background-color: transparent;
+    block-size: 100vh;
     color: $text-primary;
-    inline-size: 48rem;
-    max-block-size: 50%;
-    max-inline-size: 100%;
-    opacity: 0;
-    transform: translateY(calc(-1 * #{$spacing-06}));
-
-    /** opening and closing is used in as allow-discrete is not currently supported wide enough
-     * https://caniuse.com/mdn-css_properties_display_is_transitionable
-     */
-    transition:
-      opacity $duration-moderate-02 motion(exit, expressive),
-      transform $duration-moderate-02 motion(exit, expressive),
-      overlay $duration-moderate-02 motion(exit, expressive) allow-discrete,
-      display $duration-moderate-02 motion(exit, expressive) allow-discrete;
-
-    @media (prefers-reduced-motion) {
-      transition: none;
-    }
+    inline-size: 100vw;
+    inset-block-start: 0;
+    inset-inline-start: 0;
 
-    @include breakpoint(md) {
-      max-inline-size: 84%;
-    }
-    @include breakpoint(lg) {
-      max-inline-size: 72%;
+    &:not([open]) {
+      display: none;
     }
-    @include breakpoint(xlg) {
-      max-inline-size: 64%;
-    }
-    @include breakpoint(xlg) {
-      max-inline-size: 60%;
+
+    &::backdrop {
```

Remember to [avoid nesting selectors](https://github.com/carbon-design-system/carbon/blob/0f1e73e5777089301af89fff52f2d72a7eb2b72c/docs/style.md#L901) and instead use a fully qualified selector where possible. IIRC this was something the previous iteration of this file did fairly well.

```suggestion
  .#{$prefix}--dialog::backdrop {
```